### ISSUE TITLE
Why `<component is="constantName"/> is considered BAD?

### ISSUE THREAD

**Issue by disfated** [vuejs/eslint-plugin-vue#1136]
Concerning <https://github.com/vuejs/eslint-plugin-vue/blob/master/docs/rules/require-component-is.md>

I'm just wondering why
```html
<component is="constantName"/>
```
is considered BAD,
while
```html
<component :is="'constantName'"/>
```
is somehow GOOD?

`is` is qualified as a "special attrubute" in the [api docs](https://vuejs.org/v2/api/#is), same as `key`, which you obviously can use in a non-bound form.
Also, vue docs mention non-bound form of `is` attribute twice: [here](https://vuejs.org/v2/api/#is) and [here](https://vuejs.org/v2/guide/components.html#DOM-Template-Parsing-Caveats).

So, why one absolutely should bind `is` value to some kind of a variable, even if its value is constant, why not just use the constant directly?

**Comment by ota-meshi** [comment]
I think it should be written as follows.

```vue
<constantName />
```

So I think the following code will most likely be a typo.

```vue
<component is="constantName"/>
```

Perhaps people will the following code instead.

```vue
<component :is="constantName"/>
```

Note: If `constantName` is the vue instance property.
